src/main/java/org/jabref/gui/openoffice/DetectOpenOfficeInstallation.java

```diff
@@ -9,75 +8,66 @@
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Future;
 
-import javax.swing.BorderFactory;
-import javax.swing.JDialog;
-import javax.swing.JLabel;
-import javax.swing.JProgressBar;
-import javax.swing.SwingConstants;
-import javax.swing.SwingUtilities;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.DialogPane;
+import javafx.scene.control.ProgressIndicator;
 
-import org.jabref.Globals;
 import org.jabref.gui.DialogService;
+import org.jabref.gui.FXDialog;
 import org.jabref.gui.desktop.JabRefDesktop;
 import org.jabref.gui.desktop.os.NativeDesktop;
-import org.jabref.gui.util.BackgroundTask;
-import org.jabref.gui.util.DefaultTaskExecutor;
 import org.jabref.gui.util.DirectoryDialogConfiguration;
 import org.jabref.logic.l10n.Localization;
 import org.jabref.logic.openoffice.OpenOfficeFileSearch;
 import org.jabref.logic.openoffice.OpenOfficePreferences;
 import org.jabref.logic.util.OS;
 import org.jabref.logic.util.io.FileUtil;
+import org.jabref.model.strings.StringUtil;
 
 /**
  * Tools for automatically detecting OpenOffice or LibreOffice installations.
  */
 public class DetectOpenOfficeInstallation {
 
     private final OpenOfficePreferences preferences;
-    private final JDialog parent;
     private final DialogService dialogService;
 
-    private JDialog progressDialog;
-
-    public DetectOpenOfficeInstallation(JDialog parent, OpenOfficePreferences preferences, DialogService dialogService) {
-        this.parent = parent;
+    public DetectOpenOfficeInstallation(OpenOfficePreferences preferences, DialogService dialogService) {
         this.preferences = preferences;
         this.dialogService = dialogService;
     }
 
     public Future<Boolean> isInstalled() {
         CompletableFuture<Boolean> future = new CompletableFuture<>();
-        if (this.checkAutoDetectedPaths(preferences)) {
-            future.complete(true);
-        } else {
-            init();
-            BackgroundTask.wrap(() -> future.complete(autoDetectPaths()))
-                          .onSuccess(x -> SwingUtilities.invokeLater(progressDialog::dispose))
-                          .executeWith(Globals.TASK_EXECUTOR);
-        }
+        future.complete(autoDetectPaths());
```

Comment:
In this context, it does not make sense to use a `Future`. The code blocks until `autoDetectPaths()` completes.